lib/mail/Mail.php

```diff
@@ -556,7 +556,9 @@ public function setSubject(
      */
     public function getSubject($personalizationIndex = 0)
     {
-        return $this->personalization[$personalizationIndex]->getSubject();
+        return $this->personalization ?
```

Comment:
This change doesn't take into account setting the subject before any personalizations are added. Using the OPs example:
```
$mail = new \SendGrid\Mail\Mail();
$mail->setSubject('my subject');
$mail->getSubject();
```
The returned subject is still `null`.